arrow-avro/src/writer/format.rs

```diff
@@ -84,20 +99,31 @@ impl AvroFormat for AvroOcfFormat {
     }
 }
 
-/// Raw Avro binary streaming format (no header or footer).
+/// Raw Avro binary streaming format using **Single-Object Encoding** per record.
+///
+/// Each record written by the stream writer is framed with a prefix determined
+/// by the schema fingerprinting algorithm.
+///
+/// See: <https://avro.apache.org/docs/1.11.1/specification/#single-object-encoding>
+/// See: <https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/index.html#wire-format>
 #[derive(Debug, Default)]
-pub struct AvroBinaryFormat;
+pub struct AvroBinaryFormat {}
 
 impl AvroFormat for AvroBinaryFormat {
+    const NEEDS_PREFIX: bool = true;
     fn start_stream<W: Write>(
         &mut self,
         _writer: &mut W,
-        _schema: &Schema,
-        _compression: Option<CompressionCodec>,
+        schema: &Schema,
```

Comment:
We can probably change this back since it's still not being used.

```suggestion
        _schema: &Schema,
```